### PR TITLE
Dai semantics - Fungible Token representation of Vat state

### PR DESCRIPTION
**Pull request by ehildenb**

### CODE REVIEW

**Comment by ehildenb** [comment]
Blocked on #21

File: dai.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+            <dai-account-id> .Address </dai-account-id>
+            <dai-dai-balance> 0:Wad </dai-dai-balance>
+            <dai-allowances> .Map </dai-allowances>
+          </dai-account>
+        </dai-accounts>
+      </dai-token>
+
+    syntax DaiStep ::= "transfer" Address Wad
+ // -----------------------------------------
+    rule <k> Dai . transfer ACCOUNT_SRC AMOUNT => . ... </k>
```
Comment (ehildenb) [inline dai.md]: AMOUNT positive?

File: dai.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+           <dai-balance> BALANCE_SRC </dai-balance>
+           ...
+        </dai-account>
+      requires BALANCE_SRC >=Int AMOUNT
+
+    rule <k> Dai . transferFrom ACCOUNT_SRC ACCOUNT_DST AMOUNT => . ... </k>
+         <dai-account>
+           <dai-acct-id> ACCOUNT_SRC </dai-acct-id>
+           <dai-balance> BALANCE_SRC => BALANCE_SRC -Int AMOUNT </dai-balance>
+           <allowances> ... ACCOUNT_DST |-> (ALLOWANCE_SRC_DST => ALLOWANCE_SRC_DST -Int AMOUNT) ... </allowances>
```
Comment (ehildenb) [inline dai.md]: <dai-allowances>

File: dai.md [hunk 3]
```diff
@@ -42,31 +45,119 @@ module DAI
...
+    rule <k> Dai . transferFrom ACCOUNT_SRC ACCOUNT_DST AMOUNT => . ... </k>
+         <dai-balance>
+          ...
+          ACCOUNT_SRC |-> (BALANCE_SRC => BALANCE_SRC -Int AMOUNT)
+          ACCOUNT_DST |-> (BALANCE_DST => BALANCE_DST +Int AMOUNT)
+          ...
+        </dai-balance>
+        <dai-allowance>
+          ...
+          { ACCOUNT_SRC -> ACCOUNT_DST } |-> (ALLOWANCE_SRC_DST => ALLOWANCE_SRC_DST -Int AMOUNT)
```
Comment (gtrepta) [inline dai.md]: I don't recognize this syntax, `{ _ -> _ }`, where does this come from?

File: dai.md [hunk 4]
```diff
@@ -42,31 +45,119 @@ module DAI
...
+           ...
+         </dai-balance>
+      requires BALANCE_SRC >=Int AMOUNT
+
+    rule <k> Dai . transfer ACCOUNT_DST AMOUNT => . ... </k>
+         <msg-sender> ACCOUNT_SRC </msg-sender>
+         <dai-balance>
+           ...
+           ACCOUNT_SRC |-> (BALANCE_SRC => BALANCE_SRC -Int AMOUNT)
+           ACCOUNT_DST |-> (BALANCE_SRC => BALANCE_SRC +Int AMOUNT)
```
[the remark this answers is not in the document]
Comment (ehildenb) [reply dai.md]: Fixed.

File: dai.md (hunk 3 above)
Comment (ehildenb) [reply dai.md]: It's defined further up in the file.

**Pull request merged**